bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -133,16 +135,16 @@ def run_task_raw(task_name, targets, description = nil, task_args = nil, &block)
 
     # Wrap parameters marked with '"sensitive": true' in the task metadata with a
     # Sensitive wrapper type. This way it's not shown in logs.
-    if task.parameters
+    if (params = task['metadata']['parameters'])
       use_args.each do |k, v|
-        if task.parameters[k] && task.parameters[k]['sensitive']
+        if params[k] && params[k]['sensitive']
           use_args[k] = Puppet::Pops::Types::PSensitiveType::Sensitive.new(v)
         end
       end
     end
 
     if executor.noop
-      if task.supports_noop
+      if task['metadata']['supports_noop']
```

Comment:
Looking at this compared to the older method it seems like we'll need to spread default logic around a lot more than if "task" was a ruby object that we could add methods too.

Reply:
Yeah, I think that's the next thing I'm looking at cleaning up.